### PR TITLE
WIP: Frontend auth check

### PR DESCRIPTION
**Pull request by aapeliv**
**Backend checklist**
- [ ] Formatted my code by running `ruff check --select I --fix . && ruff check . && ruff format .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [ ] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Web frontend checklist**
- [ ] Formatted my code with `make format`
- [ ] There are no warnings from `make lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by aapeliv** [comment]
@bakeiro

File: app/web/Dockerfile [hunk 1]
```diff
@@ -16,9 +16,12 @@ RUN yarn install --frozen-lockfile
...
 COPY . .
 
-ENV PROTOS_SHA=63b57d8c12b291f8f5c1fd740879530e7da7865797f783dafcda7b3432c2efba
+# ENV PROTOS_SHA=b084f43d1becbe18a256819907a9beeb2008966d47eb9f7786aac49c6c381706
+# ENV PROTOS_URL=https://protos--by-sha.preview.couchershq.org/$PROTOS_SHA/ts.tar.gz
 
-RUN wget https://protos--by-sha.preview.couchershq.org/$PROTOS_SHA/ts.tar.gz
+ENV PROTOS_URL=https://develop--protos.preview.couchershq.org/ts.tar.gz
+
+RUN wget $PROTOS_URL
```
Comment (bakeiro) [inline app/web/Dockerfile:24]: why this? (just to understand it)

File: app/web/features/auth/useAuthStore.ts [hunk 2]
```diff
@@ -161,12 +161,50 @@ export default function useAuthStore() {
...
+        setLoading(false);
+      },
     }),
     //note: there should be no dependenices on the state or t, or
     //some useEffects will break. Eg. the token login in Login.tsx
     [setAuthenticated, setJailed, setUserId, setFlowState, queryClient]
   );
 
+  useEffect(() => {
+    if (!authenticated && typeof window !== "undefined") {
```
Comment (bakeiro) [inline app/web/features/auth/useAuthStore.ts:203]: when & how often could the variables `authActions` & `authenticated` change? this means, are we calling `checkAuthStatus` periodically?

File: app/web/Dockerfile (hunk 1 above)
Comment (aapeliv) [reply app/web/Dockerfile:24]: It's not really related to this PR: but it's a bit cumbersome every time people want to change the protos to have to change this hash. I am not sure if I want this though: I like it when they point to a particular version.

File: app/web/features/auth/useAuthStore.ts (hunk 2 above)
Comment (aapeliv) [reply app/web/features/auth/useAuthStore.ts:203]: I don't actually know how that works. I thought this meant it rusn every time `useAuthStore` is loaded?

File: app/web/Dockerfile (hunk 1 above)
Comment (bakeiro) [reply app/web/Dockerfile:24]: ✅

File: app/web/features/auth/useAuthStore.ts (hunk 2 above)
Comment (bakeiro) [reply app/web/features/auth/useAuthStore.ts:203]: like, this code runs every time the content of either `authenticated` or `authActions` change, then this code it's executed (and the first time this component is rendered).  

if this pretends to be executed periodically I don't know (by looking at the code) if this is achieved, you can use Useffect without a second parameter, so it's executed every time is redered

Comment (bakeiro) [reply app/web/features/auth/useAuthStore.ts:203]: if works it works, anyhow I can double check the behavior in my local set up

**Comment by aapeliv** [comment]
Superseded by https://github.com/Couchers-org/couchers/pull/4693

**Pull request closed without merging**